EIPS/eip-1845.md

```diff
@@ -0,0 +1,65 @@
+---
+eip: <to be assigned>
+title: Fork Name Standards
+author: DimitryKh <dimitry@ethereum.org>
+discussions-to: https://github.com/ethereum/retesteth/issues/23
+status: Draft
+category: Informational
+created: 2019-03-18
+---
+
+## Simple Summary
+Name forks like  `ETH-0001, ETH-0002, ETH-0003` and so on.
+
+## Abstract
+Each client parse `ETH-0001` kind of indexes into ethereum fork rules. `ETH-0001` stands for Frontier rules.
+Therefore if in any config file instead of `Frontier`, an `ETH-0001` name provided. The client must parse it as a `Frontier` rule.
+Parsing previous naming convention (`Frontier, Homestead, EIP150`) becomes optional. `ETH-xxxx` naming convention is mandatory for all clients to implement. 
+
+## Motivation
+**Clients:**  
+Different clients use different fork names even for the main network on their genesis config. If we want a generic genesis format, we have to use a standard for the fork names. 
+
+**Development:**  
+After `ConstantinopleFix` release, devs wanted to rename it into `Petersburg`. That require code changes which is development time and money.
+And code changes to the client might result in errors and bugs if not tested and reviewed properly.
+During hardfork development the name is not fixed, thus closer to the release date, need to change the fork name again in the code and in the test files.
+
+Following standard for the fork names will allow to hardcode fork names without issues.
+
+## Specification
+In genesis config the following fields must be understood by all clients:
+
+```
+"ETH-0001-FB": "5000",
+"ETH-0002-FB": "15000",
+...
+"ETH-xxxx-FB": "25000"
+```
+
+Where `ETH-0001-FB` stands for `Frontier` fork block number. 
+
+## Rationale
+Rationale is to get a fork name standard for generic genesis format so that RPC test tools provide clients with a uniform genesis configuration.
+It is easy to add +1 to the next fork name number rather then discuss `ConstantinopleFix` vs `Petersburg` and change the constant string everywhere: both in code and tests when we want to rename.
+
+## Backwards Compatibility
+Clients could support `Frontier, Homestead` and any other naming convention. 
+
+## Test Cases
+Retesteth will generate `genesis.json` and run all tests with this specification.
+
+## Implementation
+**Aleth:**  
+Aleth replace hardcoded variable names with an array that resolve 
+```
+chainParams().forkblocks["ETH-0002-FB"]
```

Comment:
Link to that implementation?

Reply:
its just a proposal. no implementation so far